velox/tpch/gen/dbgen/include/dbgen/dss.h

```diff
@@ -435,9 +435,21 @@ int dbg_print(int dt, FILE* tgt, void* data, int len, int eol);
 #define PR_STRT(fp) /* any line prep for a record goes here */
 #define PR_END(fp) fprintf(fp, "\n") /* finish the record here */
 #ifdef MDY_DATE
-#define PR_DATE(tgt, yr, mn, dy) sprintf(tgt, "%02d-%02d-19%02d", mn, dy, yr)
+#define PR_DATE(tgt, yr, mn, dy)                               \
+  do {                                                         \
+    int res = sprintf(tgt, "19%02ld-%02ld-%02ld", yr, mn, dy); \
```

Comment:
fmt::format would be simpler here as well.

Reply:
One doubt though, fmt::format returns a cpp string whereas tgt is supposed to be a c style null terminated string(char array) . So do you want me to use fmt::format followed by strncpy as in : 
```
#define PR_DATE(tgt, yr, mn, dy) \
do { \
std::string formatted = fmt::format("{:02d}-{:02d}-19{:02d}", mn, dy, yr); \
std::strncpy(tgt, formatted.c_str(), formatted.size() + 1); \
tgt[formatted.size() + 1] = '\0'
} while (0)
```